src/Compilers/Core/MSBuildTask/Utilities.cs

```diff
@@ -200,11 +200,16 @@ public static string GenerateFullPathToTool(string toolName)
                     if(assemblyPath != null)
                     {
                         var assemblyDirectory = Path.GetDirectoryName(assemblyPath);
-                        var toolLocalLocation = Path.Combine(assemblyDirectory, toolName);
+                        var desktopToolLocalLocation = Path.Combine(assemblyDirectory, toolName);
+                        var cliToolLocalLocation = Path.Combine(assemblyDirectory, "bincore", toolName);
```

Comment:
This can't possibly work, can it? If we're running on CoreCLR does `GetCallingAssembly` exist?

Reply:
Huh. We may want to try to test this, although it doesn't need to be in this PR.

Reply:
Just tested locally, turns out that it's in `netcoreapp2.0` but not `netcoreapp1.1`.

I'm not sure what other APIs exist for this, so I'm not sure what else we could use.

Reply:
Actually, I think that should be OK -- if we're upgrading the compiler to run on NET 4.6 and netcoreapp2.0, that means that should always be available at runtime. If it doesn't work on netcoreapp1.1 right now I'm not worried because that should be irrelevant soon.